cmd/agent/main.go

```diff
@@ -1,3 +1,52 @@
 package main
 
-func main() {}
+import (
+	"fmt"
+	"github.com/forester2k/metrics/internal/service"
+	"net/http"
+	"time"
+)
+
+const defaultReportInterval = 10
+const defaultPollInterval = 2
+const defaultHost = "localhost:8080"
+
+func poll() {
+	//fmt.Println("polling", time.Now())
+	// TODO: logic
+}
+
+func report(metric service.Saver) {
+	endpoint := "http://" + flagRunAddr + "/update" + metric.Path()
+	client := &http.Client{}
+	request, err := http.NewRequest(http.MethodPost, endpoint, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Add("Content-Type", "text/plain")
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Println("Ошибка в client.Do(request)", err)
```

Comment:
```suggestion
		return fmt.Errorf("Ошибка в client.Do(request): %w", err)
```
Из-за того, что в данном месте не организован выход из функции, агент падает по панике, пытаясь дальше вызвать `resp.Body.Close()` от resp, который равен `nil`. Проверить очень просто - достаточно запустить агента без сервера.
```
Ошибка в client.Do(request) Post "http://localhost:8080/update/gauge/someGaugeMetric/3.1": dial tcp [::1]:8080: connectex: No connection could be made because the target machine actively refused it.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x140393a11]

goroutine 37 [running]:
main.report({0x140537bf0, 0xc000102048})
        C:/work/src/github.com/forester2k/metrics/cmd/agent/main.go:33 +0x391
created by main.main in goroutine 1
        C:/work/src/github.com/forester2k/metrics/cmd/agent/main.go:46 +0x1e8
```